packages/trader/src/Modules/Trading/Components/Form/TradeParams/Turbos/barrier-selector.jsx

```diff
@@ -19,29 +19,18 @@ const BarrierSelector = ({ barrier_1, onChange, setHoveredBarrier, turbos_barrie
     const toggleBarriersTable = () => {
         setIsMobileTooltipVisible(false);
         setIsBarriersTableExpanded(!is_barriers_table_expanded);
-        setSelectedBarrier(barrier_1);
     };
 
     const onBarrierClick = barrier => {
         setHoveredBarrier(null);
         setSelectedBarrier(barrier);
-        if (!isMobile()) {
-            onChange({
-                target: {
-                    name: 'barrier_1',
-                    value: barrier,
-                },
-            });
-        }
-    };
-
-    const onButtonClick = () => {
         onChange({
             target: {
                 name: 'barrier_1',
-                value: selected_barrier,
+                value: barrier,
             },
         });
+        setIsBarriersTableExpanded(!is_barriers_table_expanded);
```

Comment:
```suggestion
        setIsBarriersTableExpanded(false);
```
@kate-deriv here it always should set it to false when a barrier is clicked, so it's better to state it explicitly